OpenMTR/Utilities/ImageUtils.cs

```diff
@@ -39,5 +39,60 @@ public static Mat GetKernel(Size size)
         {
             return Cv2.GetStructuringElement(MorphShapes.Rect, size);
         }
+
+        public static void RotateImage(Mat sourceImage, Mat destinationImage, double degrees)
+        {
+            Mat rotationMatrix = Cv2.GetRotationMatrix2D(new Point2f(sourceImage.Cols / 2, sourceImage.Rows / 2), degrees, 1);
+            Cv2.WarpAffine(sourceImage, destinationImage, rotationMatrix, sourceImage.Size());
+        }
+
+        public static float DetectImageSkew(Mat sourceImage, Mat destinationImage)
+        {
+            LineSegmentPolar[] lines = Cv2.HoughLines(destinationImage, 1, Cv2.PI / 180f, 330);
+            return FilterImageSkew(lines);
+        }
+
+        private static float FilterImageSkew(LineSegmentPolar[] lines)
+        {
+            List<LineSegmentPolar> filteredLines = new List<LineSegmentPolar>();
+
+            float theta_min = (float)(60f * Cv2.PI / 180f),
+                  theta_max = (float)(120f * Cv2.PI / 180f),
+                  theta_avr = 0f,
+                  theta_deg = 0f;
+
+            foreach (LineSegmentPolar line in lines)
+            {
+                if (line.Theta >= theta_min && line.Theta <= theta_max)
+                {
+                    filteredLines.Add(line);
+                    theta_avr += line.Theta;
+                }
+            }
+
+            if (filteredLines.Count > 0)
+            {
+                theta_avr /= filteredLines.Count;
+                theta_deg = (float)(theta_avr / Cv2.PI * 180f) - 90;
+            }
+
+            return theta_deg;
+        }
+
+        public static void DetectOdometer(Meter meter)
+        {
+            Cv2.FindContours(meter.ModifiedImage, out Point[][] contours, out HierarchyIndex[] hierarchy, RetrievalModes.External, ContourApproximationModes.ApproxSimple);
+            Rect rectangle = new Rect();
+            foreach (Point[] point in contours)
+            {
+                double area = Cv2.ContourArea(point);
+
+                if (area == 11445)
```

Comment:
Hard coding this value will be good for this one read but probably won't work for any other meters. As we move forward we'll need to find another solution.

Reply:
Agreed, for the first read, it was just easier.